### PR TITLE
test: add asset tests

### PR DESCRIPTION
**Pull request by Timeless0911**
## Summary

Add assets test:

- [x] name: [output.filename](https://rsbuild.dev/config/output/filename)
- [x]  limit: [output.dataUriLimit](https://rsbuild.dev/config/output/data-uri-limit#outputdataurilimit)
- [x] path: [output.distPath](https://rsbuild.dev/config/output/dist-path)
- [x] publicPath: [output.assetPrefix](https://rsbuild.dev/config/output/asset-prefix#outputassetprefix)
- [x] svgr: [@rsbuild/plugin-svgr](https://rsbuild.dev/plugins/list/plugin-svgr)

Some issues are not aligned or to be determined:

1. https://github.com/web-infra-dev/rslib/issues/219
2. https://github.com/web-infra-dev/rslib/issues/230
3. https://github.com/web-infra-dev/rslib/issues/231

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: e2e/cases/asset/svgr/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "asset-svgr-test",
+  "version": "1.0.0",
+  "private": true,
+  "type": "module",
+  "dependencies": {
+    "react": "^18.3.1"
```
Comment (fi3ework) [inline e2e/cases/asset/svgr/package.json:7]: should `'react'` in dev dep?

Comment (Timeless0911) [reply e2e/cases/asset/svgr/package.json:7]: react should external

Comment (fi3ework) [reply e2e/cases/asset/svgr/package.json:7]: peerDep + devDep?

Comment (fi3ework) [reply e2e/cases/asset/svgr/package.json:7]: As we usually don't make a library deps on `'react'` directly.

Comment (Timeless0911) [reply e2e/cases/asset/svgr/package.json:7]: peer is enough? We do not use react directly

**Review by fi3ework — APPROVED** [review]

**Pull request merged**